libs/types/src/dtos/graph/graph-key-pair.dto.ts

```diff
@@ -2,17 +2,24 @@ import { IsEnum, IsNotEmpty, IsString } from 'class-validator';
 import { KeyType } from './key-type.enum';
 import { HexString } from '@polkadot/util/types';
 import { ApiProperty } from '@nestjs/swagger';
+import { IsHexValue } from '#utils/decorators/is-hex-value.decorator';
 
 // DTO for the graph key pair
 export class GraphKeyPairDto {
-  @IsNotEmpty()
-  @IsString()
-  @ApiProperty({ description: 'Public graph encryption key as a hex string (prefixed with "0x")', type: String })
+  @IsHexValue({ minLength: 64, maxLength: 64, message: 'publicKey should be a 32 bytes value in hex format!' })
```

Comment:
I wonder, is it worth it to write a decorator that's able to validate that a hex string decodes to a valid public key generated by a specific algorithm?

Reply:
I have an issue for semantics checks and will add this inside that.